Shared/Supporting Files/Views/SampleList.swift

```diff
@@ -65,6 +69,12 @@ private extension SampleList {
         /// The sample displayed in the row.
         let sample: Sample
         
+        /// A string representation of the selected sample.
+        let selection: String?
+        
+        /// The current device type.
+        let deviceType = UIDevice.current.userInterfaceIdiom
```

Comment:
When it comes to UI. design, checking the user interface idiom is almost never the right choice. The reason is because of split screen on iPad. The better option is to think in terms of size classes and use the horizontal and vertical size class environment values to determine how the UI should be displayed. That will ensure that the UI looks great on not just iPhone and iPad full screen, but also on iPad split screen.

Reply:
Okay, can size classes be used to distinguish an iPad from an iPhone? Both iPhone and iPad have a compact horizontalSizeClass and regular verticalSizeClass (in the default orientation). 

There's a check on the device type since in iOS 16+ the List item selection color is the app's accent color on iPad and Mac Catalyst, and gray on iPhone.

Reply:
I didn't fully understand the problem. Almost every time I've seen a user interface idiom check, size classes should have been checked instead. This seems to be one of the rare exceptions where a user interface idiom check might be appropriate.

Reply:
I haven't tried, but does this work with iPhone 14PM horizontal orientation?

Reply:
@yo1995 Good catch. It does not. The List item selection color goes from grey to the app's accent color when switching to landscape mode.

![Simulator Screen Shot - iPhone 14 Pro Mac iOS 16 2 - 2023-04-14 at 12 09 30](https://user-images.githubusercontent.com/117859673/232135154-61f2b083-7418-435a-a183-2d0772c546e1.png)

Reply:
It sounds like checking the horizontal and vertical size class is the right approach.